### ISSUE TITLE
Validation process takes forever after 2nd epoch

### ISSUE THREAD

**Issue by KatieBelli** [ZFTurbo/Music-Source-Separation-Training#176]
**Issue:**
If I start training (or resume it) then the first epoch in that (resumed) training and the validation process work fine.

Then the second epoch in that training is done and it starts with the validation dataset.
Everything is fine until it reaches song 8 or 9 from the validation dataset. Then the remaining time suddenly slows down extremely.

_For example: The remaining time would have been at that point 5 minutes but the remaining time increases instead and the validation dataset process stops._

I tried it with steps = 10 and overlap = 1, and the validation process only took a few minutes longer after the second epoch but everything was normal again after epoch 3.

How can I solve this problem?

**My changes:**

I switched from sdr to aura_mrstft because it suits my goal alot better.

I added it in the bat file like this:

```
    --metrics aura_mrstft ^
    --metric_for_scheduler aura_mrstft ^
```

For training, I switched from masked loss to multistft_loss (adjusted this in the train python file).

I stick with higher num_steps because I had good experiences with this in the past.

Here is the config:

> audio:
>   chunk_size: 130560
>   dim_f: 1024
>   dim_t: 256
>   hop_length: 512
>   n_fft: 2048
>   num_channels: 2
>   sample_rate: 44100
>   min_mean_abs: 0.001
> 
> model:
>   act: gelu
>   bottleneck_factor: 4
>   growth: 128
>   norm: InstanceNorm
>   num_blocks_per_scale: 2
>   num_channels: 128
>   num_scales: 5
>   num_subbands: 4
>   scale:
>   - 2
>   - 2
> 
> training:
>   batch_size: 2
>   gradient_accumulation_steps: 3
>   grad_clip: 0
>   instruments:
>   - Vocals
>   - Instruments
>   lr: 1.0
>   patience: 5
>   reduce_factor: 0.95
>   target_instrument: Vocals
>   num_epochs: 400
>   num_steps: 5486
>   q: 0.95
>   coarse_loss_clip: true
>   ema_momentum: 0.999
>   optimizer: prodigy
>   other_fix: false # it's needed for checking on multisong dataset if other is actually instrumental
>   use_amp: true # enable or disable usage of mixed precision (float16) - usually it must be true
>   use_torch_checkpoint: true
> 
> augmentations:
>   enable: true # enable or disable all augmentations (to fast disable if needed)
> 
>   all:
>     channel_shuffle: 0.5 # Set 0 or lower to disable
>     random_inverse: 0.1 # inverse track (better lower probability)
>     random_polarity: 0.5 # polarity change (multiply waveform to -1)
> 
>     # pedalboard pitch shift block
>     pedalboard_pitch_shift: 0.1
>     pedalboard_pitch_shift_semitones_min: -8
>     pedalboard_pitch_shift_semitones_max: 7
> 
>     seven_band_parametric_eq: 0.25
>     seven_band_parametric_eq_min_gain_db: -5
>     seven_band_parametric_eq_max_gain_db: 5
> 
> 
> inference:
>   batch_size: 1
>   dim_t: 256
>   num_overlap: 8

GPU: NVIDIA RTX 3070 Ti
CPU: Intel i7-9700K (num_workers 8)

Edit: I checked my validation dataset wav files and none seems to be corrupted.

**Comment by lucellent** [comment]
Most likely you run out of VRAM.
After the first epoch is done, sometimes VRAM increases by 1-2 GB, so reduce chunk size/batch size.

Also overlap 8 is not really needed when training, 2 is more than enough and much faster.

**Comment by idkcode22** [comment]
> The VRAM increase after first epoch is most likely because of not cleaning perfectly the memory after validation step. I remember I've fixed this issue in the past by adding some "torch.cuda.empty_cache()" after validation is done, and maybe some gc. I don't have these edits anymore unfortunately (and it was on old version of the repo) Maybe [@kiselecheck](https://github.com/kiselecheck) can check that ?

I did that too, and it greatly improved my overall training time, especially on a 12 GB GPU
